flink-java/src/main/java/org/apache/flink/api/java/DataSet.java

```diff
@@ -1266,6 +1267,34 @@ public long count() throws Exception {
 	}
 
 	/**
+	 * Range-partitions a DataSet on the specified key fields with customed data distribution.
```

Comment:
there seems too many partionBy related function in DataSet, i suggest you add these interfaces to DataSetUtils, which is a helper class to DataSet. Besides, you should add interface for both java and scala.